coinpaprika/src/endpoint/price.ts

```diff
@@ -25,25 +20,39 @@ const convertFromTicker = async (ticker: string, coinId: string | undefined) =>
       (x: { symbol: string; rank: number }) =>
         x.symbol.toLowerCase() === ticker.toLowerCase() && x.rank !== 0,
     )
-  // if (!coin) return callback('Could not find coin', null)
+  if (typeof coin?.id === 'undefined') {
+    throw new Error('Coin id not found')
+  }
   return coin.id.toLowerCase()
 }
 
+const overrideSymbol = (overrides: Override | undefined, symbol: string): string | undefined => {
```

Comment:
Isn't this method already defined in `validator.overrideSymbol`?

Reply:
This is a special case, happens in paprika and coingecko. We don't want to get the `overrideSymbol` by default, as if doesn't exists, we want to query their ID, and for that, we need the symbol that is on `input.data`

Reply:
Because `validator.overrideSymbol` will throw? But I'm not sure that is the right thing to do. These overrides should not be required IMHO, so if we have an override we override if not we should continue as is.

Reply:
Is a special case for paprika and coingecko. If the asked symbol is not in `overrides`, we need its `id` from the data provider. For this, we need the requested symbol

Reply:
Can we make `overrideSymbol` overloaded in the validator to account for this? Should be cleaner overall.